src/controllers/controllermanager.cpp

```diff
@@ -150,10 +152,12 @@ void ControllerManager::slotShutdown() {
     locker.unlock();
 
     // Delete enumerators and they'll delete their Devices
-    foreach (ControllerEnumerator* pEnumerator, enumerators) {
+    for (ControllerEnumerator* pEnumerator: enumerators) {
         delete pEnumerator;
     }
 
+    delete m_pKeyboardController;
```

Comment:
It would be safer to set m_pKeyboardController to nullptr after deleting.

But you don't need to do that manually if you replace the  plain pointer with std::unique_ptr:
m_pKeyboardController.reset();